server/src/test/java/org/elasticsearch/transport/RemoteConnectionStrategyTests.java

```diff
@@ -208,16 +214,21 @@ public void testConnectionAttemptLogging() {
                 new ClusterConnectionManager(TestProfiles.LIGHT_PROFILE, mock(Transport.class), threadContext)
             )
         ) {
+            assert transportService.getTelemetryProvider() != null;
+            final var meterRegistry = transportService.getTelemetryProvider().getMeterRegistry();
+            assert meterRegistry instanceof RecordingMeterRegistry;
+            final var metricRecorder = ((RecordingMeterRegistry) meterRegistry).getRecorder();
```

Comment:
Related to my other comments about static fields. If we create two `TransportService` instances representing two nodes and test connection strategies for each of them, the static fields will make `RemoteConnectionStrategy` objects from different `TransportService` instance share the same metrics give incorrect results.